### PR TITLE
#P04 As Participant, I would like to see a list of any upcoming Hackathons.

### PR DESCRIPTION
**Pull request by Ri-Dearg**
Hiya! I have a draft to fix issue #21 ready to go.

**Description**
- A simple ListView that displays the Hackathon models in a card-based format. I tried to make it similar to lessons on the CI lesson pages.
- The view is paginated for every eight objects. I provided links for an organiser's page and a detail page in the future.
- The pagination buttons are also an `include` template using the default `page_obj`, so it is modular, allowing use for other paginated lists.
- I have added additional styles to `style.css` for the cards, the paginator and text colors.
- Added a draft menu item in the Navbar, "Hackathons".

**Tests**
- I wrote tests that check for the templates used, URL status, context and queryset.
- I have verified the page with the HTML5 verifier.
- Responsiveness checked using the Chrome inspector on all device sizes.

**Known Bugs**
None I can find, the view keeps it simple! 😃

Let me know what ye think, what yere vision is, I'll put in the additions to fix it right up!

**Screenshots**
Screenshot 2020-10-17 at 18 49 29
Screenshot 2020-10-17 at 18 50 28
Screenshot 2020-10-17 at 18 51 22

### CODE REVIEW

File: hackathon/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+from hackathon.models import Hackathon
+
+
+class TestHackathonViews(TestCase):
+    """Tests views for the Hackathon app."""
+
+    def setUp(self):
+        """Sets up the models for testing"""
+        user = User.objects.create(username="testuser")
+        user.save()
```
Comment (stefdworschak) [inline hackathon/tests/test_views.py:14]: I think you can remove `user.save()`. The model is saved automatically when you call `create()`. Might have overlooked that in your previous tests. If you have used the same there you can go back in another PR and change that as well.

File: static/css/style.css [hunk 2]
```diff
@@ -72,3 +72,31 @@ footer {
     color: var(--s-grey);
     background-color: var(--p-grey);
 }
+
+/* TEXT */
+.p-orange {
+   color: var(--p-orange);
```
Comment (stefdworschak) [inline static/css/style.css:78]: Make sure you keep the indentation level consistend (4 spaces). Please adjust for anything below as well

File: hackathon/templates/hackathon/includes/hackathon_card.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<article class="card shadow">
+	<div class="card-body">
+		<h5 class="p-orange card-title">{{ hackathon.display_name }}</h5>
+		<h6 class="card-subtitle mb-2 text-muted">{{ hackathon.start_date }} - {{ hackathon.end_date }}</h6>
+		<p class="card-text">{{ hackathon.description }}</p>
+        {% if hackathon.organiser %}
+		<p>Organiser: <a href="#" class="p-blue card-link">{{ hackathon.organiser }}</a></p>
```
Comment (stefdworschak) [inline hackathon/templates/hackathon/includes/hackathon_card.html:7]: Not sure if we need to show the organiser, but let's leave it for the moment.

File: hackathon/templates/hackathon/includes/paginator.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+        <ul class="pagination justify-content-center">
+            {% if page_obj.has_previous %}
```
Comment (stefdworschak) [inline hackathon/templates/hackathon/includes/paginator.html:2]: This is a really cool solution for pagination!

**Review by stefdworschak — APPROVED** [review]
LGTM

@Ri-Dearg looks great. Just finish up the few minor changes and I will merge it.

**Comment by stefdworschak** [comment]
Actually, just saw it's only a draft. But I'd say we can merge it already and you can work on the other pages in another PR.

File: hackathon/tests/test_views.py (hunk 1 above)
Comment (Ri-Dearg) [reply hackathon/tests/test_views.py:14]: Will do!

File: static/css/style.css (hunk 2 above)
Comment (Ri-Dearg) [reply static/css/style.css:78]: I'll go fix that up now

File: hackathon/templates/hackathon/includes/hackathon_card.html (hunk 3 above)
Comment (Ri-Dearg) [reply hackathon/templates/hackathon/includes/hackathon_card.html:7]: Yeah, I thought it could be handy if people had public-facing profiles. It might not be so useful otherwise.

File: hackathon/templates/hackathon/includes/paginator.html (hunk 4 above)
Comment (Ri-Dearg) [reply hackathon/templates/hackathon/includes/paginator.html:2]: Thanks!

**Review by Ri-Dearg — COMMENTED** [review]
Format changes made for `style.css` and removed `.save()`.

**Comment by stefdworschak** [comment]
@Ri-Dearg looks great! Merging now.
